src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/InvocationMessage.cs

```diff
@@ -58,12 +58,32 @@ public InvocationMessage(string invocationId, bool nonBlocking, string target, E
             Target = target;
             _arguments = arguments;
             _argumentBindingException = argumentBindingException;
+        }
+    }
+
+    public class NonStreamInvocationMessage : InvocationMessage
```

Comment:
I don't like this class name. Lets change the hierarchy a bit. I'd like InvocationMessage to be the regular invocation and we can use InvocationMessageBase as the base type.

Reply:
I originally had exactly that I did not like it because we use the base class in many places and InvocationMessage is a very good fit for the invocation message as concept. When I created NonStreamInvocationMessage I hoped someone will recommend a better name - I don't like it at all but it makes it clear what the difference between different kinds of IncocationMessages is (as opposed to InvocationMessage derived from InvocationMessageBase)

Reply:
I don't like it because InvocationMessage is the primary type. NonStreamingInvocation makes it seems like StreamingInvocation is the primary type.

Reply:
I understand that but I am more in favor of changing `NonStreamingInvocation` to something that does not make it sound secondary. How about:
- changing NonStreamingInvocationMessage to HubMethodInvocationMessage?
or changing NonStreamingInvocationMessage to HubMethodInvocationMessage and StreamingInvocationMessage to HubMethodInvocation
or changing the base class from InvocationMessage to HubMethodInvocationMessage, NonStreamingInvocationMessage to InvocationMessage and leave StreamingInvocationMessage as is? 
(I think I like the 3rd option the best)

Reply:
3rd option is best

Reply:
l'll do that then.